tools/docker/test-image/Dockerfile

```diff
@@ -1,21 +1,24 @@
-FROM ubuntu:latest
+FROM ubuntu:20.04
 
 RUN \
+        export DEBIAN_FRONTEND=noninteractive && \
         dpkg --add-architecture i386 && \
         apt-get update -qq && \
-        apt-get install -y nasm gdb unzip p7zip-full openjdk-8-jre wget python python3 qemu-system-x86 git-core build-essential libc6-dev-i386-cross libc6-dev-i386 && \
-        wget https://nodejs.org/dist/v8.9.4/node-v8.9.4-linux-x64.tar.xz && \
-        tar xfv node-v8.9.4-linux-x64.tar.xz && \
-        rm node-v8.9.4-linux-x64.tar.xz && \
+        apt-get dist-upgrade -y
+RUN \
+        export DEBIAN_FRONTEND=noninteractive && \
+        apt-get install -y nasm gdb unzip p7zip-full openjdk-8-jre wget python python3 qemu-system-x86 git-core build-essential libc6-dev-i386-cross libc6-dev-i386 clang curl
+RUN \
+        NODEVSN=v14.16.0 && \
+        wget https://nodejs.org/dist/$NODEVSN/node-$NODEVSN-linux-x64.tar.xz && \
+        tar xfv node-$NODEVSN-linux-x64.tar.xz && \
+        rm node-$NODEVSN-linux-x64.tar.xz && \
+        echo 'export PATH="$PATH:/node-$NODEVSN-linux-x64/bin"' >> ~/.bashrc && \
         wget https://sh.rustup.rs -O rustup.sh && \
         sh ./rustup.sh -y && \
         rm ./rustup.sh && \
         export PATH="$HOME/.cargo/bin:$PATH" && \
         rustup toolchain install nightly && \
         rustup default nightly && \
         rustup target add wasm32-unknown-unknown --toolchain nightly && \
-        rustup component add rustfmt-preview --toolchain nightly && \
-        apt-get clean && \
-        apt-get autoclean && \
-        apt-get autoremove && \
-        rm -rf /var/lib/apt/lists/*
+        rustup component add rustfmt-preview --toolchain nightly
```

Comment:
I'm no docker expert, but the following were intentional to keep the image small:

- Run all commands in a single `RUN` block
- Run the `apt clean` at the end.

Please let me know if there are good reasons for this change, or revert them otherwise.

Reply:
My personal practice is that what you describe is appropriate when creating docker images that then get pushed around/used in 'production' - but this seems to be an image used for development purposes, and so the ease of e.g. 
 - tweaking lines without rerunning all of the (potentially long) previous steps
 - hacking around with new packages at the end without `apt-get update`
(both of which I was personally doing) outweighed the desire for an optimally compact image.